src/Raven.Client/Documents/BulkInsert/BulkInsertOperation.cs

```diff
@@ -263,6 +269,51 @@ public BulkInsertOperation(string database, IDocumentStore store, BulkInsertOpti
 
             _generateEntityIdOnTheClient = new GenerateEntityIdOnTheClient(_requestExecutor.Conventions,
                 entity => AsyncHelpers.RunSync(() => _requestExecutor.Conventions.GenerateDocumentIdAsync(database, entity)));
+
+            _streamLock = new AsyncReaderWriterLock();
+            _lastWriteToStream = DateTime.Now;
+            _timer = new Timer( async o => await SendHeartBeat(o).ConfigureAwait(false),
```

Comment:
before doing any of that, you need to check `LastServerVersion` in request executor? What will happen if you will send a heartbeat to a server that does not support those?

Reply:
done